File: src/test/java/com/oz/office_tastezip/common/response/ResponseTest.java
package com.oz.office_tastezip.common.response;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import static org.assertj.core.api.Assertions.assertThat;

class ResponseTest {

    @Test
    @DisplayName("성공 응답 - 메시지만 있는 경우")
    void successWithoutData() {
        ResponseSuccess<Void> response = new ResponseSuccess<>();
        ResponseEntity<Response.Body<Void>> result = response.success();

        assertThat(result.getStatusCode()).isEqualTo(HttpStatus.OK);
        assertThat(result.getBody()).isNotNull();
        assertThat(result.getBody().code()).isEqualTo(ResponseCode.SUCCESS.getCode());
        assertThat(result.getBody().message()).isEqualTo(ResponseCode.SUCCESS.getMessage());
        assertThat(result.getBody().data()).isNull();
    }

    @Test
    @DisplayName("성공 응답 - 데이터 포함")
    void successWithData() {
        String data = "Hello, world!";
        ResponseSuccess<String> response = new ResponseSuccess<>();
        ResponseEntity<Response.Body<String>> result = response.success(data);

        assertThat(result.getStatusCode()).isEqualTo(HttpStatus.OK);
        assertThat(result.getBody()).isNotNull();
        assertThat(result.getBody().data()).isEqualTo(data);
    }

    @Test
    @DisplayName("성공 응답 - 커스텀 메시지 포함")
    void successWithCustomMessage() {
        String customMessage = "사용자 생성 완료";
        ResponseEntity<Response.Body<Void>> result = new ResponseSuccess<Void>(customMessage).success();

        assertThat(result.getBody()).isNotNull();
        assertThat(result.getBody().message()).isEqualTo(customMessage);
    }

    @Test
    @DisplayName("실패 응답 - 기본 메시지 사용")
    void failWithDefaultMessage() {
        ResponseEntity<Response.Body<Void>> result = new ResponseFail<Void>(ResponseCode.INTERNAL_ERROR).fail();

        assertThat(result.getStatusCode()).isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR);
        assertThat(result.getBody()).isNotNull();
        assertThat(result.getBody().code()).isEqualTo(ResponseCode.INTERNAL_ERROR.getCode());
        assertThat(result.getBody().message()).isEqualTo(ResponseCode.INTERNAL_ERROR.getMessage());
    }

    @Test
    @DisplayName("실패 응답 - 커스텀 에러 메시지")
    void failWithCustomMessage() {
        ResponseEntity<Response.Body<Void>> result = new ResponseFail<Void>(ResponseCode.TOKEN_EXPIRED, "토큰 인증 실패").fail();

        assertThat(result.getStatusCode()).isEqualTo(HttpStatus.UNAUTHORIZED);
        assertThat(result.getBody()).isNotNull();
        assertThat(result.getBody().code()).isEqualTo(ResponseCode.TOKEN_EXPIRED.getCode());
        assertThat(result.getBody().message()).isEqualTo("토큰 인증 실패");
    }
}

File: src/test/kotlin/com/oz/office_tastezip/common/response/ResponseTest.kt
package com.oz.office_tastezip.common.response

import com.oz.office_tastezip.global.response.Response
import com.oz.office_tastezip.global.response.ResponseCode
import com.oz.office_tastezip.global.response.ResponseFail
import com.oz.office_tastezip.global.response.ResponseSuccess
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import io.kotest.matchers.nulls.shouldBeNull
import io.kotest.matchers.nulls.shouldNotBeNull
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity

class ResponseTest : StringSpec({

    "성공 응답 - 메시지만 있는 경우" {
        val response = ResponseSuccess<Void>()
        val result: ResponseEntity<Response.Body<Void>> = response.success()

        result.statusCode shouldBe HttpStatus.OK
        result.body.shouldNotBeNull().let {
            it.code shouldBe ResponseCode.SUCCESS.code
            it.message shouldBe ResponseCode.SUCCESS.message
            it.data.shouldBeNull()
        }
    }

    "성공 응답 - 데이터 포함" {
        val data = "This is common response test !"
        val response = ResponseSuccess<String>()
        val result = response.success(data)

        result.statusCode shouldBe HttpStatus.OK
        result.body.shouldNotBeNull().data shouldBe data
    }

    "성공 응답 - 커스텀 메시지 포함" {
        val customMessage = "사용자 생성 완료"
        val result = ResponseSuccess<Void>(customMessage).success()

        result.body.shouldNotBeNull().message shouldBe customMessage
    }

    "실패 응답 - 기본 메시지 사용" {
        val result = ResponseFail<Void>(ResponseCode.INTERNAL_ERROR).fail()

        result.statusCode shouldBe HttpStatus.INTERNAL_SERVER_ERROR
        result.body.shouldNotBeNull().let {
            it.code shouldBe ResponseCode.INTERNAL_ERROR.code
            it.message shouldBe ResponseCode.INTERNAL_ERROR.message
        }
    }

    "실패 응답 - 커스텀 에러 메시지" {
        val result = ResponseFail<Void>(ResponseCode.TOKEN_EXPIRED, "토큰 인증 실패").fail()

        result.statusCode shouldBe HttpStatus.UNAUTHORIZED
        result.body.shouldNotBeNull().let {
            it.code shouldBe ResponseCode.TOKEN_EXPIRED.code
            it.message shouldBe "토큰 인증 실패"
        }
    }
})
